tests/uws/models_test.py

```diff
@@ -0,0 +1,455 @@
+"""Tests for the XML serialization of UWS elements"""
+
+from datetime import timezone as tz
+from unittest import TestCase
+from xml.etree.ElementTree import canonicalize
+
+from lxml import etree
+
+from vo_models.xml.generics import VODateTime
+from vo_models.xml.uws import (
+    ErrorSummary,
+    Jobs,
+    JobSummary,
+    Parameter,
+    Parameters,
+    ResultReference,
+    Results,
+    ShortJobDescription,
+)
+from vo_models.xml.uws.types import ExecutionPhase
+
+UWS_NAMESPACE_HEADER = """xmlns:uws="http://www.ivoa.net/xml/UWS/v1.0"
+xmlns:xlink="http://www.w3.org/1999/xlink"
+xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+"""
+
+# New schema versions can be downloaded from https://www.ivoa.net/xml/ under "UWS - Universal Worker Service"
+# The most current version is 1.1, found here: https://www.ivoa.net/xml/UWS/UWS-v1.1.xsd
+with open("tests/uws/UWS-Schema-V1.0.xsd", "r") as schema_file:
+    uws_schema = etree.XMLSchema(etree.parse(schema_file))
+
+
+class TestErrorSummaryType(TestCase):
+    """Tests for the UWS errorSummary complex type"""
+
+    test_error_summary_xml = f"""<uws:errorSummary {UWS_NAMESPACE_HEADER}
+    type="transient"
+    hasDetail="true">
+    <uws:message>Invalid query.</uws:message>
+    </uws:errorSummary>
+    """
+
+    def test_read_from_xml(self):
+        """Test reading from XML"""
+
+        error_summary = ErrorSummary.from_xml(self.test_error_summary_xml)
+        self.assertEqual(error_summary.type, "transient")
+        self.assertEqual(error_summary.has_detail, True)
+        self.assertEqual(error_summary.message, "Invalid query.")
+
+    def test_write_to_xml(self):
+        """Test writing to XML"""
+
+        error_summary = ErrorSummary(type="transient", has_detail=True, message="Invalid query.")
+        error_summary_xml = error_summary.to_xml(encoding=str)
+
+        self.assertEqual(
+            canonicalize(self.test_error_summary_xml, strip_text=True),
+            canonicalize(error_summary_xml, strip_text=True),
+        )
+
+class TestParameterType(TestCase):
+    """Tests for the UWS Parameter complex type"""
+
+    test_parameter_xml = (
+        f'<uws:parameter {UWS_NAMESPACE_HEADER} byReference="false" id="param1" isPost="false">'
```

Comment:
No particular reason beyond oversight. I am somewhat torn, however. I should probably change the `error_summary` one to match the rest with a series of single-lines. [PEP-8 recommends](https://www.python.org/dev/peps/pep-0008/):

> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces.

However, you are right that it makes copying it elsewhere easier- I've had to do that myself countless times. 

I think in this case I'll probably opt to use the series of single-lines. Since this is just a test package I don't really expect anyone but developers to be using this for anything more than validating their models, and in that case I think it should follow Python styling. 

I think what might be more useful is adding additional docs pages for each protocol that shows instantiating & serializing the models, with what the expected XML output should be. I think doing that will kill two birds with one stone, showing how to use them, but also giving an easy copy-paste-able output to compare, especially if that's in some cleanly selectable markdown or something.

Reply:
I have made this it's own task, as I'd like for them to be somewhat comprehensive.

https://jira.stsci.edu/browse/ASB-24561